src/main/java/jenkins/branch/BranchBuildStrategy.java

```diff
@@ -184,15 +214,47 @@ public final boolean automaticBuild(@NonNull SCMSource source,
                                         @NonNull SCMRevision currRevision,
                                         @CheckForNull SCMRevision prevRevision,
                                         @NonNull TaskListener listener) {
+        return automaticBuild(source, head, currRevision, prevRevision, prevRevision, listener);
+    }
+
+    /**
+     * API: Should the specified {@link SCMRevision} of the {@link SCMHead} for the specified {@link SCMSource} be
```

Comment:
Probably should also add a JAVADOC comment to the class warning that the SPI methods are only to be invoked through the API methods in order to allow safe evolution.

Also add a note that changing the API may require updating any SPI implementations that are also API consumers, specifically the Any, All and None implementations in basic-branch-build-strategies

Reply:
Hmmm also perhaps call it out in an update to the `implementation.adoc` https://github.com/jenkinsci/branch-api-plugin/blob/master/docs/implementation.adoc (which on initial inspection appears to be missing even a section on implementing branch build strategies.

@fbelzunc you need to document the subtleties between last built and last seen in that implementation guide. Doesn't need to be long, a sentence or two should do